src/screens/Movie/Movie.tsx

```diff
@@ -49,23 +50,31 @@ const Movie = ({ match, location }: RouteComponentProps<MovieRouteParams>): JSX.
 
   // Media
   const { isLoading, error, data: item } = useMedia(id);
-  const itemRequiresSubscription = item?.requiresSubscription !== 'false';
   useBlurImageUpdater(item);
   const { data: trailerItem } = useMedia(item?.trailerId || '');
   const { data: playlist } = useRecommendedPlaylist(recommendationsPlaylist || '', item);
 
-  const isFavorited = useFavoritesStore((state) => !!item && state.hasItem(item));
+  // AccessModel & entitlement
+  const mediaOffer = useMemo(() => (item?.productIds && filterCleengMediaOffers(item.productIds)) || undefined, [item]);
+  useEntitlement(mediaOffer);
 
+  const isFavorited = useFavoritesStore((state) => !!item && state.hasItem(item));
   const watchHistoryItem = useWatchHistoryStore((state) => item && state.getItem(item));
   const progress = watchHistoryItem?.progress;
 
   // General state
   const [hasShared, setHasShared] = useState<boolean>(false);
   const [playTrailer, setPlayTrailer] = useState<boolean>(false);
 
-  // User
+  // User, accessModel, entitlement
   const { user, subscription } = useAccountStore(({ user, subscription }) => ({ user, subscription }), shallow);
-  const allowedToWatch = isAllowedToWatch(accessModel, !!user, itemRequiresSubscription, !!subscription);
+  const isItemFree = item?.requiresSubscription === 'false' || !!item?.free;
+
+  const mediaOffers = useMemo(() => filterCleengMediaOffers(item?.productIds), [item]);
```

Comment:
Instead of adding another util method, would it make sense to just put all of this logic into `isAllowedToWatch`?  Is there a case where we would use the filter method on its own?

Reply:
Yes, I agree

Reply:
I'm using the result of `filterCleengMediaOffers()` for the `useEntitlement()` hook, so I can't move it to `isAllowedToWatch()`. However, I could move both their logic to useEntitlement. In that case, the hook would handle all accessModels.

Reply:
The idea is that it would all go into `isAllowedToWatch` which I guess would make it into a hook itself.  Otherwise, `isAllowedToWatch` and `filterCleengMediaOffers` could probably become methods internal to the useEntitlements hook.  Basically these 3 things will always be used together, so they can all live in 1 place and ideally only expose what external callers will need.

Reply:
Yes, I'm implementing that. Only `isAllowedToWatch` is also used for the lock icon in other places, but I'm creating a dedicated `showLock()` util for that now.